extensions/amp-selector/1.0/amp-selector.js

```diff
@@ -302,24 +301,25 @@ function OptionShim(props) {
   }, [domElement, isDisabled]);
 
   useLayoutEffect(() => {
-    domElement.setAttribute('role', role);
+    if (role) {
```

Comment:
We should not need to check role if it has a default value.

Reply:
Same issue as #29293. I'll update the type for now to avoid propagating these ugly `if`s.